### PR TITLE
fix(idn): encode url except domain

### PR DESCRIPTION
**Pull request by curbengh**
Compatibility with hexo v4 (https://github.com/hexojs/hexo/pull/3708), while maintaining backward compatibility with v3, #74 is not compatible with v3 nor IDN.
Fixes https://github.com/hexojs/hexo-generator-feed/issues/63

Supersede #66 

Fixes #74

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -67,4 +68,16 @@ describe('Sitemap generator', () => {
       result.should.be.ok;
     });
   });
+
+  it('IDN handling', () => {
+    hexo.config.url = 'http://fôo.com/bár';
+    const parsedUrl = encodeURL(hexo.config.url);
+
+    const result = generator(locals);
+    const $ = cheerio.load(result.data);
```
Comment (curbengh) [inline test/index.js]: `result.data` still show "http://yoursite.com/". Same issue as https://github.com/hexojs/hexo-generator-sitemap/pull/66/files#r314965773

**Comment by tomap** [comment]
Your title is wrong. Idn should not be encoded to punycode. Ever. Just the path should be encoded.

**Comment by curbengh** [comment]
> Your title is wrong. Idn should not be encoded to punycode. Ever. Just the path should be encoded.

Updated title and commit message.

IDN can be encoded to punycode, but I agree it's not a requirement.

File: test/index.js (hunk 1 above)
Comment (curbengh) [reply test/index.js]: #70 resolve this issue, but brings another issue.

**Comment by curbengh** [comment]
Pending https://github.com/hexojs/hexo/pull/3708

File: test/index.js (hunk 1 above)
Comment (curbengh) [reply test/index.js]: Resolved

**Comment by curbengh** [comment]
Fixed failing test. Ready to be reviewed.

**Comment by curbengh** [comment]
@hexojs/core 
Ready for review.

**Pull request merged**